apps/api/src/server/migration/migrators/cleanup/htmlTemplates.js

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * @param {NodeJS.ReadableStream} readableStream
+ * @returns {Promise<string>}
+ */
+export const streamToString = async (readableStream) => {
+	return new Promise((resolve, reject) => {
+		/** @type {string[]} */
+		const chunks = [];
+		readableStream.on('data', (chunk) => {
+			chunks.push(chunk.toString());
+		});
+		readableStream.on('end', () => {
+			resolve(chunks.join(''));
+		});
+		readableStream.on('error', reject);
+	});
+};
+
+/**
+ *
+ * @param {string} string
+ */
+export const stringToStream = (string) => {
+	const readable = new Readable({
+		read() {
+			readable.push(string);
+			readable.push(null);
+		}
+	});
+	return readable;
+};
+
+/**
+ *
+ * @param {number} caseId
+ */
+export const getHtmlDocumentVersions = async (caseId) => {
+	const response = await getDocumentsInCase(caseId, 'text/html', 1, 9999);
+	return response.items;
+};
+
+const getBlobClient = () => {
+	// return BlobStorageClient.fromUrl(config.blobStorageUrl)
+	return BlobStorageClient.fromUrl('https://back-office-applications-docs-test.planninginspectorate.gov.uk')
+}
+/**
+ *
+ * @param {string} blobName
+ */
+export const downloadHtmlBlob = async (blobName) => {
+	const blobClient = getBlobClient();
+	const downloadBlockBlobResponse = await blobClient.downloadStream(
+		privateBlobContainer,
+		blobName.replace('/', '')
+	);
+	if (!downloadBlockBlobResponse.readableStreamBody) {
+		throw Error('No file stream received after attempting to download from Blob Container');
+	}
+	return streamToString(downloadBlockBlobResponse.readableStreamBody);
+};
+
+/**
+ *
+ * @param {string} htmlString
+ */
+export const processHtml = (htmlString) => {
+	if (!(htmlString.includes('youtube') || htmlString.includes('youtu.be'))) {
+		throw Error('No youtube link found in HTML');
+	}
+
+	const youtubeUrl = extractYouTubeURLFromHTML(htmlString);
+	return renderYouTubeTemplate(youtubeUrl);
+};
+
+/**
+ *
+ * @param {number} applicationId
+ * @param {string} documentGuid
+ * @param {*} versionProperties
+ */
+export const addDocumentVersion = (applicationId, documentGuid, versionProperties) => {
+  return createDocumentVersion(versionProperties, applicationId, documentGuid);
+};
+
+/**
+ *
+ * @param {string} blobName
+ * @param {string} htmlAsString
+ */
+export const uploadNewFileVersion = async (blobName, htmlAsString) => {
+	const blobClient = getBlobClient();
+	const htmlAsStream = stringToStream(htmlAsString);
+	const blobNameWithIncrementedVersionNumber = incrementBlobVersionNumber(blobName);
+	await blobClient.uploadStream(
+		privateBlobContainer,
+		htmlAsStream,
+		blobNameWithIncrementedVersionNumber.replace('/', ''),
+		'text/html'
+	);
+	return {
+		newBlobName: blobNameWithIncrementedVersionNumber,
+		blobSize: Buffer.from(htmlAsString, 'utf-8').length
+	};
+};
+
+/**
+ *
+ * @param {string} blobName
+ */
+export const incrementBlobVersionNumber = (blobName) => {
+	const lastColonIndex = blobName.lastIndexOf(':');
```

Comment:
may need to consider paths which are in the following format (typical cbos uploads)
`/application/BC010006/13b75263-b90b-4f1d-80df-15f6acb93fb2/2`